lib/semantics/check-select-stmt.cpp

```diff
@@ -0,0 +1,114 @@
+//===-- lib/semantics/check-select-stmt.cpp -------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-select-stmt.h"
+#include "flang/common/Fortran.h"
+#include "flang/common/idioms.h"
+#include "flang/parser/message.h"
+#include "flang/parser/tools.h"
+#include "flang/semantics/tools.h"
+#include <list>
+#include <optional>
+#include <set>
+#include <tuple>
+#include <variant>
+
+namespace Fortran::semantics {
+
+void SelectStmtChecker::Leave(
+    const parser::SelectRankConstruct &selectRankConstruct) {
+  const auto &selectRankStmt{
+      std::get<parser::Statement<parser::SelectRankStmt>>(
+          selectRankConstruct.t)};
+  const auto &selectRankStmtSel{
+      std::get<parser::Selector>(selectRankStmt.statement.t)};
+
+  // R1149 select-rank-stmt checks
+  const Symbol *saveSelSymbol{nullptr};
+  if (const auto resolvedSel{ResolveSelector(selectRankStmtSel)}) {
+    if (const Symbol * sel{evaluate::UnwrapWholeSymbolDataRef(*resolvedSel)}) {
+      if (!IsAssumedRankArray(*sel)) {
+        context_.Say(parser::FindSourceLocation(selectRankStmtSel),
+            "Selector is not an assumed-rank array variable"_err_en_US);
```

Comment:
Can you emit the name of the symbol as part of the message, here and at line 42?